launch/launch/actions/execute_process.py

```diff
@@ -686,6 +702,11 @@ async def __execute_process(self, context: LaunchContext) -> None:
                 ),
             )
 
+        if self.__shutdown_received:
+            # Shutdown received, do not run process
+            return
+        self.__running = True
```

Comment:
I would add this at the beginning of the method.

Reply:
Disregard this.

Reply:
Why? Actually @hidmic is suggesting me to change it to after the try block, and I think that would be a better idea, to avoid having the shutdown process to something that's not actually running.